File: 12 interval/interval.cpp
#include <iostream>
#include<iomanip>
using namespace std;

int main(){
	
	int n,a,b;
	double x,proizvod = 1; // производ иницијализујемо на неутрал за множење
	bool indikator = false; //индикатор нам служи да проверимо да ли се неки број налази у траженом интервалу 
                                 //  и на почетку претпоставимо да немамо бројева у траженом интервалу

	cin >> n;
	cin >> a;
	cin >> b;

//користимо петљу да учитамо n броја
	for(int i = 1; i <= n; i++){

		 //у свакој итерацији петље учитамо један број
		cin >> x;

		//ако је учитани број у траженом интервалу онда га додамо у производ
		if(x >= a && x <= b){
			proizvod *= x;

			 //променимо вредност индикатора
			if(indikator == false)
			  indikator = true;
		}
	}	
	
	 //ако немамо ниједан  број у траженом интервалу
	if(indikator == false)
		cout << "NEMA BROJEVA U DATOM INTERVALU" << endl;
	else
		cout << fixed<< showpoint << setprecision(3)<< proizvod << endl;
	return 0;
}

File: 12 interval/interval.py

n = int(input())
a = int(input())
b = int(input())

# ïğîèçâîä èíèöè¼àëèçó¼åìî íà íåóòğàë çà ìíîæåœå
proizvod = 1 
#èíäèêàòîğ íàì ñëóæè äà ïğîâåğèìî äà ëè ñå íåêè áğî¼ íàëàçè ó òğàæåíîì èíòåğâàëó è íà ïî÷åòêó ïğåòïîñòàâèìî äà íåìàìî áğî¼åâà ó òğàæåíîì èíòåğâàëó
indikator = False

#êîğèñòèìî ïåòšó äà ó÷èòàìî n áğî¼à
for i in range(n):

	#ó ñâàêî¼ èòåğàöè¼è ïåòšå ó÷èòàìî ¼åäàí áğî¼
	x = float(input())
	
	#àêî ¼å ó÷èòàíè áğî¼ ó òğàæåíîì èíòåğâàëó îíäà ãà äîäàìî ó ïğîèçâîä
	if(x >= a and x <= b):
		proizvod *= x

		#ïğîìåíèìî âğåäíîñò èíäèêàòîğà
		if (indikator == False):
			indikator = True

if (indikator == False):
	print ("NEMA BROJEVA U DATOM INTERVALU")

else:
	print ("%.3f"%proizvod)